internal/provider/repository_branch_protection_resource_test.go

```diff
@@ -0,0 +1,191 @@
+package provider_test
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/knownvalue"
+	"github.com/hashicorp/terraform-plugin-testing/statecheck"
+	"github.com/hashicorp/terraform-plugin-testing/tfjsonpath"
+)
+
+func TestAccRepositoryBranchProtectionResource(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read testing
+			{
+				Config: providerConfig + `
+resource "forgejo_repository" "test" {
+	name = "test_repo_branch_protection"
+}
+resource "forgejo_repository_branch_protection" "test" {
+	branch_name   = "main"
+	repository_id = forgejo_repository.test.id
+}
+`,
+				ConfigStateChecks: []statecheck.StateCheck{
+					statecheck.ExpectKnownValue("forgejo_repository_branch_protection.test", tfjsonpath.New("branch_name"), knownvalue.StringExact("main")),
+				},
+			},
+			// ImportState testing
+			{
+				ResourceName:                         "forgejo_repository_branch_protection.test",
+				ImportStateId:                        "tfadmin/test_repo_branch_protection/main",
+				ImportState:                          true,
+				ImportStateVerify:                    true,
+				ImportStateVerifyIdentifierAttribute: "branch_name",
+			},
+			// Update and Read testing
```

Comment:
Since the `branch_name` has a `RequiresReplace()` plan modifier, this is actually a "Recreate and Read testing". You're not including any "Update and Read testing" cases. Please add some, for typical update operations...